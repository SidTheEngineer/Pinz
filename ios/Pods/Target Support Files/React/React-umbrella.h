#ifdef __OBJC__
#import <UIKit/UIKit.h>
#endif

#import "RCTAssert.h"
#import "RCTBridge+Private.h"
#import "RCTBridge.h"
#import "RCTBridgeDelegate.h"
#import "RCTBridgeMethod.h"
#import "RCTBridgeModule.h"
#import "RCTBundleURLProvider.h"
#import "RCTConvert.h"
#import "RCTDefines.h"
#import "RCTDisplayLink.h"
#import "RCTErrorCustomizer.h"
#import "RCTErrorInfo.h"
#import "RCTEventDispatcher.h"
#import "RCTFrameUpdate.h"
#import "RCTImageSource.h"
#import "RCTInvalidating.h"
#import "RCTJavaScriptExecutor.h"
#import "RCTJavaScriptLoader.h"
#import "RCTJSStackFrame.h"
#import "RCTKeyCommands.h"
#import "RCTLog.h"
#import "RCTModuleData.h"
#import "RCTModuleMethod.h"
#import "RCTMultipartDataTask.h"
#import "RCTMultipartStreamReader.h"
#import "RCTNullability.h"
#import "RCTParserUtils.h"
#import "RCTPerformanceLogger.h"
#import "RCTRootView.h"
#import "RCTRootViewDelegate.h"
#import "RCTRootViewInternal.h"
#import "RCTTouchEvent.h"
#import "RCTTouchHandler.h"
#import "RCTURLRequestDelegate.h"
#import "RCTURLRequestHandler.h"
#import "RCTUtils.h"
#import "RCTWebSocketProxy.h"
#import "RCTWebSocketProxyDelegate.h"
#import "CSSLayout.h"
#import "CSSMacros.h"
#import "CSSNodeList.h"
#import "RCTJSCErrorHandling.h"
#import "RCTJSCExecutor.h"
#import "RCTJSCWrapper.h"
#import "RCTAccessibilityManager.h"
#import "RCTAlertManager.h"
#import "RCTAppState.h"
#import "RCTAsyncLocalStorage.h"
#import "RCTClipboard.h"
#import "RCTDevLoadingView.h"
#import "RCTDevMenu.h"
#import "RCTEventEmitter.h"
#import "RCTExceptionsManager.h"
#import "RCTI18nManager.h"
#import "RCTI18nUtil.h"
#import "RCTKeyboardObserver.h"
#import "RCTRedBox.h"
#import "RCTSourceCode.h"
#import "RCTStatusBarManager.h"
#import "RCTTiming.h"
#import "RCTUIManager.h"
#import "RCTFPSGraph.h"
#import "RCTJSCProfiler.h"
#import "RCTMacros.h"
#import "RCTProfile.h"
#import "RCTActivityIndicatorView.h"
#import "RCTActivityIndicatorViewManager.h"
#import "RCTAnimationType.h"
#import "RCTAutoInsetsProtocol.h"
#import "RCTBorderDrawing.h"
#import "RCTBorderStyle.h"
#import "RCTComponent.h"
#import "RCTComponentData.h"
#import "RCTConvert+CoreLocation.h"
#import "RCTConvert+MapKit.h"
#import "RCTDatePicker.h"
#import "RCTDatePickerManager.h"
#import "RCTFont.h"
#import "RCTMap.h"
#import "RCTMapAnnotation.h"
#import "RCTMapManager.h"
#import "RCTMapOverlay.h"
#import "RCTModalHostView.h"
#import "RCTModalHostViewController.h"
#import "RCTModalHostViewManager.h"
#import "RCTNavigator.h"
#import "RCTNavigatorManager.h"
#import "RCTNavItem.h"
#import "RCTNavItemManager.h"
#import "RCTPicker.h"
#import "RCTPickerManager.h"
#import "RCTPointerEvents.h"
#import "RCTProgressViewManager.h"
#import "RCTRefreshControl.h"
#import "RCTRefreshControlManager.h"
#import "RCTRootShadowView.h"
#import "RCTScrollableProtocol.h"
#import "RCTScrollView.h"
#import "RCTScrollViewManager.h"
#import "RCTSegmentedControl.h"
#import "RCTSegmentedControlManager.h"
#import "RCTShadowView.h"
#import "RCTSlider.h"
#import "RCTSliderManager.h"
#import "RCTSwitch.h"
#import "RCTSwitchManager.h"
#import "RCTTabBar.h"
#import "RCTTabBarItem.h"
#import "RCTTabBarItemManager.h"
#import "RCTTabBarManager.h"
#import "RCTTextDecorationLineType.h"
#import "RCTView.h"
#import "RCTViewControllerProtocol.h"
#import "RCTViewManager.h"
#import "RCTWebView.h"
#import "RCTWebViewManager.h"
#import "RCTWrapperViewController.h"
#import "UIView+Private.h"
#import "UIView+React.h"
#import "RCTActionSheetManager.h"
#import "RCTLocationObserver.h"
#import "RCTGIFImageDecoder.h"
#import "RCTImageBlurUtils.h"
#import "RCTImageCache.h"
#import "RCTImageEditingManager.h"
#import "RCTImageLoader.h"
#import "RCTImageStoreManager.h"
#import "RCTImageUtils.h"
#import "RCTImageView.h"
#import "RCTImageViewManager.h"
#import "RCTLocalAssetImageLoader.h"
#import "RCTResizeMode.h"
#import "RCTLinkingManager.h"
#import "RCTDataRequestHandler.h"
#import "RCTFileRequestHandler.h"
#import "RCTHTTPRequestHandler.h"
#import "RCTNetInfo.h"
#import "RCTNetworking.h"
#import "RCTNetworkTask.h"
#import "RCTSettingsManager.h"
#import "RCTRawTextManager.h"
#import "RCTShadowRawText.h"
#import "RCTShadowText.h"
#import "RCTText.h"
#import "RCTTextField.h"
#import "RCTTextFieldManager.h"
#import "RCTTextManager.h"
#import "RCTTextSelection.h"
#import "RCTTextView.h"
#import "RCTTextViewManager.h"
#import "RCTVibration.h"
#import "RCTSRWebSocket.h"
#import "RCTWebSocketExecutor.h"
#import "RCTWebSocketManager.h"
#import "RCTWebSocketModule.h"

FOUNDATION_EXPORT double ReactVersionNumber;
FOUNDATION_EXPORT const unsigned char ReactVersionString[];

