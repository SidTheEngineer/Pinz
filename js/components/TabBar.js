import React from 'react';
import { StyleSheet } from 'react-native';
import ScrollableTabView, { DefaultTabBar } from 'react-native-scrollable-tab-view';
import Posts from '../components/Posts';
import Map from './Map';

const styles = StyleSheet.create({
  textStyle: {
    fontSize: 16,
    color: 'white'
  },

  underlineStyle: {
    backgroundColor: 'white'
  }
});

const TabBar = props => (
  <ScrollableTabView
    renderTabBar={() => <DefaultTabBar />}
    tabBarPosition="bottom"
    tabBarBackgroundColor={'#05006d'}
    tabBarTextStyle={styles.textStyle}
    tabBarUnderlineStyle={styles.underlineStyle}
  >
    <Map {...props} tabLabel="Map" />
    <Posts tabLabel="Posts" />
  </ScrollableTabView>
);

export default TabBar;
