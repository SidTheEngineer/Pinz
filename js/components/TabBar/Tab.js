import React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

import { COLORS } from '../../constant';

const styles = StyleSheet.create({
  tab: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    width: 50,
    backgroundColor: 'rgb(245,245,245)',
    elevation: 8
  }
});

const Tab = props => (
  <TouchableOpacity
    style={styles.tab}
    onPress={() => props.goToPage(props.index)}
    activeOpacity={1}
  >
    <Icon
      name={props.tab}
      size={20}
      color={props.activeTab === props.index ? COLORS.LIGHT_GREEN : 'gray'}
    />
  </TouchableOpacity>
);

Tab.propTypes = {
  tab: React.PropTypes.string,
  goToPage: React.PropTypes.func,
  index: React.PropTypes.number,
  activeTab: React.PropTypes.number
};

export default Tab;
