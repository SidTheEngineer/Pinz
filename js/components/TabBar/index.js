import React from 'react';
import { View, StyleSheet } from 'react-native';
import ScrollableTabView from 'react-native-scrollable-tab-view';
import Tab from './Tab';
import Map from '../Map';
import CategoryList from '../CategoryList';

const styles = StyleSheet.create({
  tabContainer: {
    height: 45,
    flexDirection: 'row',
  }
});

const CustomTabBar = props => (
  // Props here are coming from the ScrollableTabView component.
  <View style={styles.tabContainer}>
    {props.tabs.map((tab, index) => (
      <Tab
        {...props}
        index={index}
        tab={tab}
        key={tab}
      />
    ))}
  </View>
);

const TabBar = props => (
  <ScrollableTabView
    tabBarPosition="bottom"
    renderTabBar={() => <CustomTabBar />}
    locked
  >
    <Map tabLabel="map" {...props} />
    <CategoryList tabLabel="list" />
  </ScrollableTabView>
);

CustomTabBar.propTypes = {
  tabs: React.PropTypes.oneOfType([
    React.PropTypes.arrayOf(React.PropTypes.string),
    React.PropTypes.string
  ])
};

export default TabBar;
