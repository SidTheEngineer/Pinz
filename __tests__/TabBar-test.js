import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TabBar from '../js/components/TabBar';

jest.mock('react-native-scrollable-tab-view', () => 'ScrollableTabView');
jest.mock('react-native-maps', () => 'MapView');

// The TabBar is responsible for returning a MapView, so
// react-native-maps is needed, as well as the initialRegion as a prop.
const mapProps = {
  initialRegion: {
    latitude: 28.601660,
    longitude: -81.200788,
    latitudeDelta: 0.0206,
    longitudeDelta: 0.014
  }
};

describe('TabBar', () => {
  it('renders correctly', () => {
    const tabBar = renderer.create(
      <TabBar map={mapProps} />
    ).toJSON();
    expect(tabBar).toMatchSnapshot();
  });
});
