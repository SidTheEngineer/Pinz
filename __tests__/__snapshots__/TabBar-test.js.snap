exports[`TabBar renders correctly 1`] = `
<ScrollableTabView
  locked={true}
  renderTabBar={[Function]}
  tabBarPosition="bottom">
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      }
    }>
    <MapView
      loadingEnabled={true}
      onRegionChange={[Function]}
      pitchEnabled={false}
      provider="google"
      region={
        Object {
          "latitude": 28.60166,
          "latitudeDelta": 0.0206,
          "longitude": -81.200788,
          "longitudeDelta": 0.014,
        }
      }
      rotateEnabled={false}
      showsCompass={false}
      style={
        Object {
          "flex": 5,
        }
      } />
  </View>
  <ScrollView
    style={
      Object {
        "flex": 1,
      }
    }>
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "borderBottomWidth": 0.5,
          "borderColor": "rgb(150,150,150)",
          "borderStyle": "solid",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "height": 60,
        }
      }>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "mistyrose",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "margin": 5,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          TEST
        </Text>
      </View>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "borderBottomWidth": 0.5,
          "borderColor": "rgb(150,150,150)",
          "borderStyle": "solid",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "height": 60,
        }
      }>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "mistyrose",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "margin": 5,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          TEST
        </Text>
      </View>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "borderBottomWidth": 0.5,
          "borderColor": "rgb(150,150,150)",
          "borderStyle": "solid",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "height": 60,
        }
      }>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "mistyrose",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "margin": 5,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          TEST
        </Text>
      </View>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "borderBottomWidth": 0.5,
          "borderColor": "rgb(150,150,150)",
          "borderStyle": "solid",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "height": 60,
        }
      }>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "mistyrose",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "margin": 5,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          TEST
        </Text>
      </View>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "borderBottomWidth": 0.5,
          "borderColor": "rgb(150,150,150)",
          "borderStyle": "solid",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "height": 60,
        }
      }>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "mistyrose",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "margin": 5,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          TEST
        </Text>
      </View>
    </TouchableOpacity>
  </ScrollView>
</ScrollableTabView>
`;
